src/include/duckdb/function/udf_function.hpp

```diff
@@ -0,0 +1,317 @@
+																	   input.size(),
+																	   udf_func);
+									};
+	    RegisterFunction(name, args, ret_type, udf_function);
+	}
+
+	template<typename TR, typename... Args>
+	void CreateBinaryFunction(string name, vector<SQLType> args, SQLType ret_type, TR (*udf_func)(Args...)) {
+		assert(sizeof...(Args) == 2);
+		CreateBinaryFunction<TR, Args...>(name, args, ret_type, udf_func);
+	}
+
+	template<typename TR, typename TA, typename TB>
+	void CreateBinaryFunction(string name, vector<SQLType> args, SQLType ret_type, TR (*udf_func)(TA, TB)) {
+		if(args.size() != 2) {
+			throw duckdb::Exception(ExceptionType:: MISMATCH_TYPE,
+									"The number of SQLType arguments (\"args\") should be 2!");
+		}
+		if(!TypesMatch<TA>(args[0])) {
+			throw duckdb::Exception(ExceptionType::MISMATCH_TYPE, "The first arguments don't match!");
+		}
+		if(!TypesMatch<TB>(args[1])) {
+			throw duckdb::Exception(ExceptionType::MISMATCH_TYPE, "The second arguments don't match!");
+		}
+
+		udf_function_t udf_function = [=] (DataChunk &input, ExpressionState &state, Vector &result) {
+										BinaryExecutor::Execute<TA, TB, TR>(input.data[0],
+																			input.data[1],
+																			result,
+																			input.size(),
+																			udf_func);
+									};
+		RegisterFunction(name, args, ret_type, udf_function);
+	}
+
+	template<typename TR, typename... Args>
+	void CreateTernaryFunction(string name, vector<SQLType> args, SQLType ret_type, TR (*udf_func)(Args...)) {
+		assert(sizeof...(Args) == 3);
+		CreateTernaryFunction<TR, Args...>(name, args, ret_type, udf_func);
+	}
+
+	template<typename TR, typename TA, typename TB, typename TC>
+	void CreateTernaryFunction(string name, vector<SQLType> args, SQLType ret_type, TR (*udf_func)(TA, TB, TC)) {
+		if(args.size() != 3) {
+			throw duckdb::Exception(ExceptionType:: MISMATCH_TYPE,
+									"The number of SQLType arguments (\"args\") should be 3!");
+		}
+		if(!TypesMatch<TA>(args[0])) {
+			throw duckdb::Exception(ExceptionType::MISMATCH_TYPE, "The first arguments don't match!");
+		}
+		if(!TypesMatch<TB>(args[1])) {
+			throw duckdb::Exception(ExceptionType::MISMATCH_TYPE, "The second arguments don't match!");
+		}
+		if(!TypesMatch<TC>(args[2])) {
+			throw duckdb::Exception(ExceptionType::MISMATCH_TYPE, "The third arguments don't match!");
+		}
+
+	    udf_function_t udf_function = [=] (DataChunk &input, ExpressionState &state, Vector &result) -> void {
+										TernaryExecutor::Execute<TA, TB, TC, TR>(input.data[0],
+																		   	   	 input.data[1],
+																		   	   	 input.data[2],
+																				 result,
+																				 input.size(),
+																				 udf_func);
+									};
+		RegisterFunction(name, args, ret_type, udf_function);
+	}
+
+	void RegisterFunction(string name, vector<SQLType> args, SQLType ret_type, udf_function_t udf_function) {
+		ScalarFunction scalar_function = ScalarFunction(name, args, ret_type, nullptr, false,
```

Comment:
This function is almost the same as the other `RegisterFunction` (signature `void RegisterFunction(string name, udf_function_t udf_function)`) above. The function above should call this function, rather than implementing the same thing twice.